lib/influxdb/query/core.rb

```diff
@@ -11,8 +11,42 @@ def version
         ping.header['x-influxdb-version']
       end
 
+      def quote(param)
+        if param.is_a?(String)
+          "'" + param.gsub(/['"\\\x0]/, '\\\\\0') + "'"
+        elsif param.is_a?(Integer) || param.is_a?(Float) || param == true || param == false
+          param.to_s
+        else
+          throw "Unexpected parameter type #{p.class} (#{p.inspect})"
+        end
+      end
+
+      def query_builder(query, opts = {})
+        if opts[:params].is_a?(Array)
+          # convert array to hash
+          params = opts[:params].each_with_object({}).with_index do |(param, hash), i|
+            hash[(i + 1).to_s.to_sym] = quote(param)
+          end
+        elsif opts[:params].is_a?(Hash)
+          params = opts[:params].each_with_object({}) do |(k, v), hash|
+            hash[k.to_sym] = quote(v)
+          end
+        elsif opts[:params].nil?
+          params = {}
+        else
+          raise ArgumentError, "Unsupported #{opts[:params].class} params"
+        end
+        begin
+          query % params
+        rescue KeyError => e
+          raise ArgumentError, e.to_s
+        end
+      end
+
       # rubocop:disable Metrics/MethodLength
       def query(query, opts = {})
+        query = query_builder(query, opts)
```

Comment:
Since we're not using any other option than `:params` in `query_builder`, we can call it directly as `query_builder(query, opts[:params])`. That eliminates the need to call `opts[:params]` in every condition and branch above.